plugins/factory-plugin/src/theia-commands.ts

```diff
@@ -117,6 +144,50 @@ export class TheiaCloneCommand {
 
 }
 
+export class TheiaZipCloneCommand implements TheiaCloneCommand {
+
+    private locationURI: string | undefined;
+    private folder: string;
+    private zipfile: string;
+
+    constructor(project: cheApi.workspace.ProjectConfig | cheApi.workspace.devfile.Project, projectsRoot: string) {
+        if (isDevfileProjectConfig(project)) {
+            const source = project.source;
+
+            this.locationURI = source.location;
+            this.folder = path.join(projectsRoot, project.name);
+            const zipfileRE = /[^\/]+(?:\.zip)$/i;
+            const match = this.locationURI.match(zipfileRE);
+            this.zipfile = match ? match[0] : `${project.name}.zip`;
+        } else {
+            // legacy project config
+            theia.window.showErrorMessage('Legacy workspace config is not supported. Please use devfile instead');
+        }
+    }
+
+    execute(): PromiseLike<void> {
+        if (!this.locationURI) {
+            return new Promise(() => { });
+        }
+
+        const clone = async (progress: theia.Progress<{ message?: string; increment?: number }>, token: theia.CancellationToken): Promise<void> => {
+            try {
+                const zipfilePath = await zip.execWget(this.locationURI, this.folder, this.zipfile);
+                await zip.execZip(this.folder, zipfilePath);
+            } catch (e) {
+                theia.window.showErrorMessage(`Couldn't download ${this.locationURI}: ${e.message}`);
```

Comment:
Maybe for error we should use token: theia.CancellationToken to stop progressor?

Reply:
AFAIK, cancellation token is used to check or get notified when long-running command is cancelled by user.